pkg/metrics/metrics.go

```diff
@@ -23,23 +23,30 @@ import (
 )
 
 // StartMetrics starts the server based on the metricsConfig provided by the user.
-func StartMetrics(config metricsConfig) {
-	// Register metrics only when the metric list is provided by the operator
-	if config.collectorList != nil {
-		RegisterMetrics(config.collectorList)
+func StartMetrics(config metricsConfig) error {
+	if config.metricsRegisterer == nil {
+		config.metricsRegisterer = prometheus.DefaultRegisterer
+		config.metricsGatherer = prometheus.DefaultGatherer
 	}
-
-	http.Handle(config.metricsPath, promhttp.Handler())
+	err := RegisterMetrics(config.metricsRegisterer, config.collectorList)
```

Comment:
What happens if the collectorList is empty?

Reply:
Nothing, because there is a for loop which ranges overs the list. And ranging over an empty list is a no-op.

Reply:
Does a distinction between `[]` and `nil` matter here, or are they both treated the same by RegisterMetrics?

Reply:
They would both be the same because of the for loop.